src/Definition.php

```diff
@@ -6,59 +6,48 @@
  * @license   https://github.com/laminas/laminas-server/blob/master/LICENSE.md New BSD License
  */
 
+declare(strict_types=1);
+
 namespace Laminas\Server;
 
 use Countable;
 use Iterator;
+use Laminas\Server\Exception\InvalidArgumentException;
 
-/**
- * Server methods metadata
- */
 class Definition implements Countable, Iterator
 {
     /**
-     * @var array Array of \Laminas\Server\Method\Definition objects
+     * @var Method\Definition[]
      */
     protected $methods = [];
 
     /**
-     * @var bool Whether or not overwriting existing methods is allowed
+     * @var bool
      */
     protected $overwriteExistingMethods = false;
 
-    /**
-     * Constructor
-     *
-     * @param  null|array $methods
-     */
-    public function __construct($methods = null)
+    public function __construct(?array $methods = null)
```

Comment:
note for later: we might want to change it to `iterable` type, instead of array

Reply:
Agreed.